doc-files/release-notes-21.md

```diff
@@ -0,0 +1,152 @@
+# Release Notes for JavaFX 21
+
+## Introduction
+
+The following notes describe important changes and information about this release. In some cases, the descriptions provide links to additional detailed information about an issue or a change.
+
+These release notes cover the standalone JavaFX 21 release. JavaFX 21 requires JDK 17 or later.
+
+## Important Changes
+
+### JavaFX Requires macOS 11 or Later
+
+On Mac platforms, JavaFX 21 requires macOS 11 or later. An exception will be thrown when initializing the JavaFX runtime on older versions of macOS.
```

Comment:
just curious: all these lists (CSRs, Enhancements, Bug Fixes) are compiled from JBS, correct?

Reply:
Yes, although there is no list of CSRs. The first section includes issues tagged with "release-note=yes". The list of enhancements and bugs are from the following JBS filter:

[FX 21 RN Issues](https://bugs.openjdk.org/issues/?filter=44607)

Reply:
Why are [8305885](https://bugs.openjdk.org/browse/JDK-8305885) and [8304290](https://bugs.openjdk.org/browse/JDK-8304290) not included in release notes?

Are issues JDK-8301022, JDK-8301712, JDK-8302294, JDK-8302684, JDK-8303217, JDK-8303748 not included because they were introduced and fixed in the same version?

Lastly, I was of the opinion that backports are not a part of the release notes, but it seems that [JDK-8313324](https://bugs.openjdk.org/browse/JDK-8313324) is still added?

Reply:
> Why are [8305885](https://bugs.openjdk.org/browse/JDK-8305885) and [8304290](https://bugs.openjdk.org/browse/JDK-8304290) not included in release notes?

They were in the initial list I had, and then I went through and excluded a number of cleanup issues. I guess I excluded too many. I'll add these two back in.

> Are issues JDK-8301022, JDK-8301712, JDK-8302294, JDK-8302684, JDK-8303217, JDK-8303748 not included because they were introduced and fixed in the same version?

Yes, except for [JDK-8303748](https://bugs.openjdk.org/browse/JDK-8303748) which I excluded because it is a build issue that isn't relevant to application developers.

> Lastly, I was of the opinion that backports are not a part of the release notes, but it seems that [JDK-8313324](https://bugs.openjdk.org/browse/JDK-8313324) is still added?

Prior to this release, we wouldn't have included backports. However, with the change to the stabilization process such that all fixes first go into master and are then backported to the stabilization release, we would miss actual fixes that are first delivered in this release if we didn't consider them. Note that we need to use the base bug ID for backports that were first fixed in jfx21.

Reply:
> Note that we need to use the base bug ID for backports that were first fixed in jfx21

Can you shed some light on this for future reference?

In case of [JDK-8313324](https://bugs.openjdk.org/browse/JDK-8313324), we do not use the base bug id. The same was also applied to jfx21.0.1, should we be using [JDK-8313331](https://bugs.openjdk.org/browse/JDK-8313331) or the base bug id for 21.0.1 release notes?

Reply:
We should _always_ use the base bug ID and _never_ the backport ID, which for this bug is [JDK-8313227](https://bugs.openjdk.org/browse/JDK-8313227).

I will look into adjusting the JBS filter to make this a less error-prone process (using the `masterbug` macro which will provide the base bug IDs for a given set of backports).

Reply:
> We should always use the base bug ID and never the backport ID, which for this bug is [JDK-8313227](https://bugs.openjdk.org/browse/JDK-8313227).

And to be clear I meant that the "base bug ID" for this bug is [JDK-8313227](https://bugs.openjdk.org/browse/JDK-8313227).

Reply:
my bad, the release notes has the correct base bug id 👍 

my local jfx21 release notes skipped [JDK-8313227](https://bugs.openjdk.org/browse/JDK-8313227) as it was labelled as `hgupdate-sync`.

Reply:
This will need some careful JBS query logic when I update it to properly handle backports. The `hgupdate-sync` label is specific to the particular record (backport or main bug) for a particular code line. So the `hgupdate-sync` label on the main bug mean that the bug should be skipped, but only for jfx22 (by virtue of the fact that it is in jfx21).